checker/tests/run-pass/precondition.rs

```diff
@@ -11,11 +11,11 @@ extern crate mirai_annotations;
 
 pub fn main() {
     let mut a = [1, 2];
-    foo(&mut a, 3); //~ unsatisfied precondition
+    foo(&mut a, 3); //~ unsatisfied precondition: i should be 0 or 1
 }
 
 fn foo(arr: &mut [i32; 2], i: usize) {
-    precondition!(i < 2); //~ related location
+    precondition!(i < 2, "i should be 0 or 1"); //~ related location
```

Comment:
Using the first expression as the default message might be nice:

  precondition!(i < 2);

would print

  unsatisfied precondition: i < 2

just a thought.

Reply:
It may well be, but doing so in the obvious way does not seem to work.